common/lib/types/stats.ts

```diff
@@ -0,0 +1,308 @@
+		this.webhook = new MessageTypes(values && values.webhook);
+		this.sharedQueue = new MessageTypes(values && values.sharedQueue);
+		this.externalQueue = new MessageTypes(values && values.externalQueue);
+		this.httpEvent = new MessageTypes(values && values.httpEvent);
+		this.push = new MessageTypes(values && values.push);
+		this.all = new MessageTypes(values && values.all);
+	}
+}
+
+class MessageDirections {
+	all?: MessageTypes;
+	inbound?: MessageTraffic;
+	outbound?: MessageTraffic;
+
+	constructor(values?: MessageDirectionsValues) {
+		this.all           = new MessageTypes(values && values.all);
+		this.inbound       = new MessageTraffic(values && values.inbound);
+		this.outbound      = new MessageTraffic(values && values.outbound);
+	}
+}
+
+class XchgMessages {
+	all?: MessageTypes;
+	producerPaid?: MessageDirections;
+	consumerPaid?: MessageDirections;
+
+	constructor(values?: XchgMessagesValues) {
+		this.all           = new MessageTypes(values && values.all);
+		this.producerPaid  = new MessageDirections(values && values.producerPaid);
+		this.consumerPaid  = new MessageDirections(values && values.consumerPaid);
+	}
+}
+
+class PushStats {
+	messages?: number;
+	notifications?: NotificationsValues;
+	directPublishes?: number;
+
+	constructor(values?: PushValues) {
+		this.messages = (values && values.messages) || 0;
+		let notifications = values && values.notifications;
+		this.notifications = {
+			invalid: notifications && notifications.invalid || 0,
+			attempted: notifications && notifications.attempted || 0,
+			successful: notifications && notifications.successful || 0,
+			failed: notifications && notifications.failed || 0
+		};
+		this.directPublishes = (values && values.directPublishes) || 0;
+	}
+}
+
+class ProcessedCount {
+	succeeded?: number;
+	skipped?: number;
+	failed?: number;
+
+	constructor(values: ProcessedCountValues) {
+		this.succeeded = (values && values.succeeded) || 0;
+		this.skipped = (values && values.skipped) || 0;
+		this.failed = (values && values.failed) || 0;
+	}
+}
+
+class ProcessedMessages {
+	delta?: Record<string, ProcessedCount>;
+
+	constructor(values?: ProcessedMessagesValues) {
+		this.delta = undefined;
+		if (values && values.delta) {
+			this.delta = { };
+			forInOwnNonNullProps(values.delta, (prop: string) => {
+				(this.delta as Record<string, ProcessedCount>)[prop] = new ProcessedCount((values.delta as Record<string, ProcessedCountValues>)[prop]);
+			});
+		}
+	}
+}
+
+class Stats extends MessageDirections {
+	persisted?: MessageTypes;
+	connections?: ConnectionTypes;
+	channels?: ResourceCount;
+	apiRequests?: RequestCount;
+	tokenRequests?: RequestCount;
+	xchgProducer?: XchgMessages;
+	xchgConsumer?: XchgMessages;
+	push?: PushStats;
+	processed?: ProcessedMessages;
+	inProgress?: never;
+	unit?: never;
+	intervalId?: never;
+
+	constructor(values?: StatsValues) {
+		super(values as MessageDirectionsValues);
+		this.persisted     = new MessageTypes(values && values.persisted);
```

Comment:
Variable spacing within statements to form columns like this is generally not a good idea as it will cause bigger than necessary diff noise when lines get added or removed from this 'bag of setters'.

Reply:
Yeah I'm a bit unsure about how to proceed with this (and the other styling differences you mentioned above). I actually don't think it's necessarily a bad thing for us to conform these files again in future using `prettier`. The alternative would be to reformat every file as I change them which would massively increase the burden on the reviewer to understand what's changed in each PR. I absolutely agree that this TypeScript migration is the right time to introduce prettier to the repo but my intention at the moment is to do so at the end.

Reply:
Fair enough and thanks for explaining. I can see we have #684 in the backlog, supported by #685. In terms of code quality, implementing both of those will raise the game massively and will also reduce the need for code review comments like the one I made to start this conversation even exist. 😄